components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/HandlerManager.java

```diff
@@ -0,0 +1,193 @@
+/*
+ *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.auth.service.handler;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.core.bean.context.MessageContext;
+import org.wso2.carbon.identity.core.handler.HandlerComparator;
+import org.wso2.carbon.identity.core.handler.IdentityHandler;
+import org.wso2.carbon.identity.core.handler.IdentityMessageHandler;
+import org.wso2.carbon.identity.core.handler.MessageHandlerComparator;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.sort;
+
+/**
+ * HandlerManager class can be used to get the correct handlers,
+ * just before execute it either list or first priority one.
+ */
+public class HandlerManager {
+
+    private static Log log = LogFactory.getLog(HandlerManager.class);
+    private static HandlerManager handlerManager = new HandlerManager();
+
+    private HandlerManager(){
+
+    }
+
+    public static HandlerManager getInstance() {
+        if (log.isDebugEnabled()) {
+            log.debug("Created singleton instance for " + HandlerManager.class.getName());
+        }
+        return HandlerManager.handlerManager;
+    }
+
+    /**
+     * Get the first priority handler after sort and filter the enabled handlers.
+     *
+     * @param identityHandlers
+     * @param isEnableHandlersOnly
+     * @return IdentityHandler
+     */
+    public <T extends IdentityHandler> T getFirstPriorityHandler(List<T> identityHandlers,
+                                                                 boolean isEnableHandlersOnly) {
```

Comment:
Yes that should be a typo :-) Fixed it with the suggested name 'checkEnabledHandlerOnly'